### DISCUSSION TITLE
能否在umi4中修改reactDom的挂载根组件的方式

### DISCUSSION THREAD

**Discussion by chaserzz** [umijs/umi#11954] [Q&A]
因为一些库中的setState的使用方式还是17的版本的使用方式，但是需要使用到这些库的内容。想要知道能否在umi4中有办法能使用老的ReactDom.render的方法去实现根组件的挂载

**Comment by GuanJdoJ** [comment]
`rootContainer` 运行时修改交给 react-dom 渲染时的根组件 [文档-rootContainer](https://umijs.org/docs/api/runtime-config#rootcontainerlastrootcontainer-args)
<img width="683" alt="image" src="https://github.com/umijs/umi/assets/30386473/dae1ddec-b3b9-4550-8137-5b4181d56dc4">

**Comment by fz6m** [comment]
umi 4 是支持 react 16 / 17 / 18 的，只要在你的 `package.json` 中显示安装你想要的版本的 `react` / `react-dom` 就可以，react dom 的挂载方法会自动根据版本来改变，这样你就能用 react 17 了；如果没显示安装，默认使用 react 18 。
